python/pyarrow/_parquet_encryption.pxd

```diff
@@ -131,3 +131,22 @@ cdef extern from "arrow/python/parquet_encryption.h" \
             SafeGetFileDecryptionProperties(
             const CKmsConnectionConfig& kms_connection_config,
             const CDecryptionConfiguration& decryption_config)
+
+cdef extern from "arrow/dataset/parquet_encryption_config.h" namespace "arrow::dataset" nogil:
+    cdef cppclass CParquetEncryptionConfig "arrow::dataset::ParquetEncryptionConfig":
+        CParquetEncryptionConfig() except +
+        void Setup(shared_ptr[CCryptoFactory] crypto_factory,
+                   shared_ptr[CKmsConnectionConfig] kms_connection_config,
+                   shared_ptr[CEncryptionConfiguration] encryption_config)
+
+    cdef cppclass CParquetDecryptionConfig "arrow::dataset::ParquetDecryptionConfig":
+        CParquetDecryptionConfig() except +
+        void Setup(shared_ptr[CCryptoFactory] crypto_factory,
+                   shared_ptr[CKmsConnectionConfig] kms_connection_config,
+                   shared_ptr[CDecryptionConfiguration] decryption_config)
+
+
+cdef public shared_ptr[CCryptoFactory] pyarrow_unwrap_cryptofactory(object crypto_factory)
```

Comment:
It doesn't seem necessary to create and expose public Cython/C APIs for this. We could just add the necessary Cython declarations for the caller code to unwrap those objects directly, like we do for example here:
https://github.com/apache/arrow/blob/ad7f6ef3a97be6a3eb6fe37d0df1b1634db057d2/python/pyarrow/_parquet.pxd#L550-L564